pkg/kubelet/kubelet_network.go

```diff
@@ -222,6 +226,41 @@ func (kl *Kubelet) reconcileCBR0(podCIDR string) error {
 	return nil
 }
 
+const (
+	// Maximum number of retries of node status update.
+	updateNodeStatusMaxRetries int = 3
+)
+
+// Clear NetworkUnavailable condition when the network plugin signals it is
+// ready by not indicating an error from its Status() method
+func (kl *Kubelet) clearNetworkingCondition() {
+	if kl.kubeClient == nil {
+		return
+	}
+
+	var err error
+	for i := 0; i < updateNodeStatusMaxRetries; i++ {
+		// Patch could also fail, even though the chance is very slim. So we still do
+		// patch in the retry loop.
+		err = nodeutil.SetNodeCondition(kl.kubeClient, kl.nodeName, api.NodeCondition{
```

Comment:
The kubernetes network driver currently clears the condition. For kubenet, it will clear it when the node has a PodCIDR.  For CNI, it will clear when a CNI config has been correctly read from disk.  Note that this currently does *not* require the CNI plugin itself (eg, weave, calico, openshift-sdn, etc) to clear the condition.  We probably want to do that in the future, but we need to work out how that gets handled by simple CNI plugins (like kubenet, eventually) that don't have a long-running process or anything watching the apiserver that could clear the condition.

So for CNI plugins, nothing changes yet.